### PR TITLE
kv: add error code for package kv

### PR DESCRIPTION
**Pull request by jackysp**
### What problem does this PR solve? 
Not all the errors in kv package have error code.

### What is changed and how it works?
Add them. Related to https://github.com/pingcap/parser/pull/584.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported variable/fields change

Side effects

 - Breaking backward compatibility

### CODE REVIEW

**Comment by jackysp** [comment]
This PR fixes two issues. One is to separate the error that was originally mapped to ErrUnknown 1105 into multiple error codes, and the other is to put https://github.com/pingcap/tidb/pull/12866/files#diff -7e048d4e2f4a505912ffd33bd900b6d3R43 is changed to ErrDupUnknownInIndex. The first one I think is ok, the second one, I think it needs to review. PTAL @zimulala @crazycs520

File: ddl/rollingback.go [hunk 1]
```diff
@@ -40,7 +40,7 @@ func convertAddIdxJob2RollbackJob(t *meta.Meta, job *model.Job, tblInfo *model.T
 	}
 
 	if kv.ErrKeyExists.Equal(err) {
-		return ver, kv.ErrKeyExists.GenWithStack("Duplicate for key %s", indexInfo.Name.O)
+		return ver, ErrDupUnknownInIndex.GenWithStackByArgs(indexInfo.Name.O)
```
Comment (jackysp) [inline ddl/rollingback.go]: @zimulala @crazycs520 PTAL

Comment (zimulala) [reply ddl/rollingback.go]: Why change this error code? `ErrDupUnknownInIndex` error code is 1859.
```
mysql> select * from t;
+------+
| a    |
+------+
|    1 |
|    1 |
+------+
2 rows in set (0.01 sec)
mysql> alter table t add unique index idx(a);
ERROR 1062 (23000): Duplicate entry '1' for key 'idx'
```

Comment (jackysp) [reply ddl/rollingback.go]:
```suggestion
		return ver, ErrKeyExists.GenWithStackByArgs("", indexInfo.Name.O)
```
how about this?

Comment (jackysp) [reply ddl/rollingback.go]: PTAL @zimulala

File: store/tikv/snapshot.go [hunk 2]
```diff
@@ -376,7 +376,7 @@ func extractKeyErr(keyErr *pb.KeyError) error {
 		return newWriteConflictError(keyErr.Conflict)
 	}
 	if keyErr.Retryable != "" {
-		return kv.ErrTxnRetryable.FastGenByArgs("tikv restarts txn: " + keyErr.GetRetryable())
+		return kv.ErrTxnRetryable.FastGenByArgs(keyErr.GetRetryable())
 	}
 	if keyErr.Abort != "" {
 		err := errors.Errorf("tikv aborts txn: %s", keyErr.GetAbort())
```
Comment (SunRunAway) [inline store/tikv/snapshot.go]: Should this error have a error code?

Comment (jackysp) [reply store/tikv/snapshot.go]: Yes, but not in this plan. There are many similar problems.

**Pull request merged**